data/categoryRepository.go

```diff
@@ -444,64 +444,136 @@ func (r *CategoryRepository) GetCategoriesByName(name string) (categories []mode
 	if err != nil {
 		return
 	}
 	for rows.Next() {
-		var childId int64
-		err = rows.Scan(&childId)
+		dataPortalCategory := models.Category{}
+		category := models.CategoryWithAncestry{}
+		err = rows.Scan(
+			&dataPortalCategory.Id,
+			&dataPortalCategory.Universe,
+			&dataPortalCategory.Name,
+			&dataPortalCategory.IsHeader,
+			&category.Ancestry,
+			&category.DefaultGeoHandle,
+			&category.DefaultGeoFIPS,
+			&category.DefaultGeoName,
+			&category.DefaultGeoShortName,
+			&category.DefaultFrequency,
+		)
 		if err != nil {
 			return
 		}
-		children = append(children, childId)
+		if category.Ancestry.Valid {
+			dataPortalCategory.ParentId = getParentId(category.Ancestry)
+		}
+		if category.DefaultFrequency.Valid || category.DefaultGeoHandle.Valid || category.ObservationStart.Valid || category.ObservationEnd.Valid {
+			// Only initialize Defaults struct if any defaults values are available
+			dataPortalCategory.Defaults = &models.CategoryDefaults{}
+		}
+		if category.DefaultFrequency.Valid {
+			dataPortalCategory.Defaults.Frequency = &models.DataPortalFrequency{
+				Freq: category.DefaultFrequency.String,
+				Label: freqLabel[category.DefaultFrequency.String],
+			}
+		}
+		if category.DefaultGeoHandle.Valid {
+			dataPortalCategory.Defaults.Geography = &models.DataPortalGeography{
+				Handle: category.DefaultGeoHandle.String,
+			}
+			if category.DefaultGeoFIPS.Valid {
+				dataPortalCategory.Defaults.Geography.FIPS = category.DefaultGeoFIPS.String
+			}
+			if category.DefaultGeoName.Valid {
+				dataPortalCategory.Defaults.Geography.Name = category.DefaultGeoName.String
+			}
+			if category.DefaultGeoShortName.Valid {
+				dataPortalCategory.Defaults.Geography.ShortName = category.DefaultGeoShortName.String
+			}
+		}
+		children = append(children, dataPortalCategory)
 	}
 	return
 }
 
-func (r *CategoryRepository) CreateCategoryPackage(
+func (r *CategoryRepository) getCategoryTree(
 	id int64,
 	geo string,
 	freq string,
 	seriesRepository *SeriesRepository,
-) (pkg models.DataPortalCategoryPackage, err error) {
-
-	kidIds, err := r.GetChildrenOf(id)
+) (tree []models.CategoryWithInflatedSeries, err error) {
+	kids, err := r.GetChildrenOf(id)
 	if err != nil {
 		return
 	}
-	var universe string
-	for _, kidId := range kidIds {
-		inflatedCat := models.CategoryWithInflatedSeries{}
-		category, anErr := r.GetCategoryById(kidId)
-		if anErr != nil {
-			err = anErr
-			return
-		}
-		inflatedCat.Category = category
+	for _, kid := range kids {
+		if kid.IsHeader {
+			tree = append(tree, models.CategoryWithInflatedSeries{Category: kid})
 
-		if geo != "" && freq != "" {
-			seriesList, anErr := seriesRepository.GetInflatedSeriesByGroupGeoAndFreq(kidId, geo, freq, Category)
+			subtree, anErr := r.getCategoryTree(kid.Id, geo, freq, seriesRepository) //recursion
+			if anErr != nil {
+				return
+			}
+			for _, cat := range subtree {
+				tree = append(tree, cat)
+			}
+		} else {
```

Comment:
we could use `continue` to use the early return pattern and avoid some of the indentation here.